platform/core/src/services/ViewportGridService/ViewportGridService.ts

```diff
@@ -96,6 +96,10 @@ class ViewportGridService {
     this.serviceImplementation._reset();
   }
 
+  public onModeExit(): void {
```

Comment:
The reset wasn't being called because it wasn't defined as a function in the API, causing the flash of the old instance (this is the ultimate cause of the overall issue, everything else is related but additional).